src/pipecat/processors/frameworks/rtvi.py

```diff
@@ -143,29 +194,116 @@ class RTVIConfig(BaseModel):
 #
 
 
+# deprecated
 class RTVIUpdateConfig(BaseModel):
+    """DEPRECATED.
+
+    Pipeline Configuration has been removed as part of the RTVI protocol 1.0.0.
+    Use custom client and server messages instead.
+    """
+
     config: List[RTVIServiceConfig]
     interrupt: bool = False
 
 
 class RTVIActionRunArgument(BaseModel):
+    """DEPRECATED.
+
+    Actions have been removed as part of the RTVI protocol 1.0.0.
+    Use custom client and server messages instead.
+    """
+
     name: str
     value: Any
 
 
 class RTVIActionRun(BaseModel):
+    """DEPRECATED.
+
+    Actions have been removed as part of the RTVI protocol 1.0.0.
+    Use custom client and server messages instead.
+    """
+
     service: str
     action: str
     arguments: Optional[List[RTVIActionRunArgument]] = None
 
 
 @dataclass
 class RTVIActionFrame(DataFrame):
+    """DEPRECATED.
+
+    Actions have been removed as part of the RTVI protocol 1.0.0.
+    Use custom client and server messages instead.
+    """
+
     rtvi_action_run: RTVIActionRun
     message_id: Optional[str] = None
 
 
+class RTVIRawClientMessageData(BaseModel):
+    """Data structure expected from client messages sent to the RTVI server."""
+
+    t: str
+    d: Optional[Any] = None
+
+
+class RTVIClientMessage(BaseModel):
+    """Cleansed data structure for client messages for handling."""
+
+    msg_id: str
+    type: str
+    data: Optional[Any] = None
+
+
+@dataclass
+class RTVIClientMessageFrame(SystemFrame):
+    """A frame for sending messages from the client to the RTVI server.
+
+    This frame is meant for custom messaging from the client to the server
+    and expects a server-response message.
+    """
+
+    msg_id: str
+    type: str
+    data: Optional[Any] = None
+
+
+@dataclass
+class RTVIServerResponseFrame(SystemFrame):
+    """A frame for sending messages from the client to the RTVI server.
+
+    This frame is meant for custom messaging from the client to the server
+    and expects a server-response message.
+    """
+
+    client_msg: RTVIClientMessageFrame
+    data: Optional[Any] = None
+    error: Optional[str] = None
+
+
+class RTVIRawServerResponseData(BaseModel):
+    """Data structure for server responses to client messages."""
+
+    t: str
+    d: Optional[Any] = None
+
+
+class RTVIServerResponse(BaseModel):
+    """A response message from the client to the RTVI server.
```

Comment:
I think this docstring is backwards (both sentences). Implies that the server sends the initial message and the client responds.